flycheck.el

```diff
@@ -9326,39 +9326,46 @@ This syntax checker requires Rust 1.15 or newer.  See URL
   :error-explainer flycheck-rust-error-explainer
   :modes rust-mode
   :predicate flycheck-buffer-saved-p
-  :enabled (lambda ()
-             (-when-let (file (buffer-file-name))
-               (locate-dominating-file file "Cargo.toml")))
-  :verify (lambda (_)
-            (-when-let (file (buffer-file-name))
-              (let* ((has-toml (locate-dominating-file file "Cargo.toml"))
-                     (valid-crate-type (flycheck-rust-valid-crate-type-p
-                                        flycheck-rust-crate-type))
-                     (need-binary-name
-                      (and flycheck-rust-crate-type
-                           (not (string= flycheck-rust-crate-type "lib")))))
-                (list
-                 (flycheck-verification-result-new
-                  :label "Cargo.toml"
-                  :message (if has-toml "Found" "Missing")
-                  :face (if has-toml 'success '(bold warning)))
-                 (flycheck-verification-result-new
-                  :label "Crate type"
-                  :message (if valid-crate-type
-                               (format "%s" flycheck-rust-crate-type)
-                             (format "%s (invalid, should be one of 'lib', 'bin', 'test', 'example' or 'bench')"
-                                     flycheck-rust-crate-type))
-                  :face (if valid-crate-type 'success '(bold error)))
-                 (flycheck-verification-result-new
-                  :label "Binary name"
-                  :message (cond
-                            ((not need-binary-name) "Not required")
-                            ((not flycheck-rust-binary-name) "Required")
-                            (t (format "%s" flycheck-rust-binary-name)))
-                  :face (cond
-                         ((not need-binary-name) 'success)
-                         ((not flycheck-rust-binary-name) '(bold error))
-                         (t 'success))))))))
+  :enabled
+  (lambda ()
+    (and buffer-file-name
+         (locate-dominating-file buffer-file-name "Cargo.toml")))
+  :working-directory
+  (lambda (_)
+    (and buffer-file-name
+         (locate-dominating-file buffer-file-name "Cargo.toml")))
+  :verify
+  (lambda (_)
+    (and buffer-file-name
```

Comment:
This top-level and doesn't look right to me: it will cause `C-c ! v` to fail if there's no buffer-file-name.  Is that what we want? (Maybe it's an uncommon enough case that we're OK with it?)

Reply:
It won't fail `verify-setup`; there just won't be any verification message:

```
  rust-cargo
    - may enable: Automatically disabled!
    - predicate:  nil
    - executable: Found at /usr/sbin/cargo
```

We could also emit a `verification-result` indicating that the checker cannot work without a filename, but I don't think it's common enough to be worth the trouble.

Reply:
@cpitclaudel Is this good for you, or should I add a verification-result?

Reply:
Woops, sorry for missing the original message (and for the confusion!). Yes, it's great :)